map.go

```diff
@@ -335,13 +359,19 @@ func (m *Map) Close() error {
 		return nil
 	}
 
-	runtime.SetFinalizer(m, nil)
-	return unix.Close(int(m.fd))
+	return m.fd.close()
 }
 
-// FD gets the raw fd value of Map
+// FD gets the file descriptor of the Map.
+//
+// This function will panic if used after calling Close.
 func (m *Map) FD() int {
-	return int(m.fd)
+	fd, err := m.fd.value()
+	if err != nil {
+		panic(err)
```

Comment:
Discussed in person, I think we shouldn't panic() but return an `err` instead - panic'ing feels like a last resort.

Reply:
It returns `-1` now, which is essentially the same thing that the standard library does for `File.Fd`. It's highly unlikely that -1 (all ones unsigned) is ever a valid fd.